### ISSUE TITLE
Implement app shortcuts

### ISSUE THREAD

**Issue by n1snt** [organicmaps/organicmaps#4899]
**Is your feature request related to a problem? Please describe.**
Currently, the organic maps app does not support app shortcuts (refer to the shortcut below) 
![image](https://user-images.githubusercontent.com/46300244/229299963-898c38f1-a9aa-4f49-926a-cc734bd54455.png)

**Describe the solution you would like**
I think implementing shortcuts for the search & bookmarks lists would be a really good idea. It would allow users to directly go to the bookmarks list screen or search screen.

**Additional context**
Google maps currently have the following 2 shortcuts:
![image](https://user-images.githubusercontent.com/46300244/229300963-08334a61-973d-491e-a3a0-9106deaa67d3.png)

**Comment by Jean-BaptisteC** [comment]
https://developer.android.com/develop/ui/views/launch/shortcuts/creating-shortcuts
I guess we can implement this feature only with shortcuts.xml (without Java code)

**Comment by n1snt** [comment]
Yeah, but for bookmarks, it would require java code because it's a fragment.

**Comment by n1snt** [comment]
This is going to be a part of my proposal so I would love to get assigned for this issue.

**Comment by pabloscloud** [comment]
> Yeah, but for bookmarks, it would require java code because it's a fragment.

Just home and work would be sufficient for me

**Comment by n1snt** [comment]
How about adding an option in the bookmark to enable/disable shortcuts for a specific bookmark?

**Comment by biodranik** [comment]
That is an interesting idea.

**Comment by n1snt** [comment]
We can show a popup asking the user "Do you want to add this bookmark to your home screen?" and then use [pinned shortcuts](https://developer.android.com/develop/ui/views/launch/shortcuts/creating-shortcuts#pinned) to add it as a shortcut.
I'm currently on the move, I'll try to develop a mock implementation when I return home.

**Comment by ProgramminCat** [comment]
Any progress on this issue?

**Comment by millerGrey** [comment]
Hello. I made pinned shortcuts for bookmarks and tracks. It's look like this

<img width="720" height="1280" alt="Image" src="https://github.com/user-attachments/assets/79ea1acc-267b-4892-9413-53dc067bf249" />

<img width="720" height="1280" alt="Image" src="https://github.com/user-attachments/assets/86c0031b-0613-490e-8db7-90002a53bd01" />

But I have a question. It needs uid for pinned shortcuts. User can't make a dublicate of shorcut with it. I use id of bookmark or track for it. But when I restart app, ids of my bookmarks and tracks not equals with previous ones. And because of it I can make another shortcut for same bookmark. Is this normal behavior? If so, what I could take for shortcut's id?

Does this feature still need?

**Comment by biodranik** [comment]
- This feature may be useful for some frequently used shortcuts
- Unique IDs for bookmarks and tracks are planned as a part of the core refactoring, needed also for backup sync
- What is your current UI/UX for pinning bookmarks? Providing an option to pin any bookmark (except Home/Work) looks like a rare feature. A more useful shortcut could be: start a generic search, route to home, show a specific area on the map, etc.

**Comment by millerGrey** [comment]
UI for adding shortcut <img width="720" height="1280" alt="Image" src="https://github.com/user-attachments/assets/04bee853-16e7-453f-b8b2-eed490693eb7" />

I think there is no need pinned shortcut for search. Static shortcut menu will be enough for it. And what do you mean about route to home? I thought pinned shortcut open a bookmark (home for example), so user can build a route. Isn't so?
